File: face/geomajas-face-dojo/client/src/main/resources/geomajas/action/toolbar/ZoomOutAction.js
/*
 * This is part of Geomajas, a GIS framework, http://www.geomajas.org/.
 *
 * Copyright 2008-2011 Geosparc nv, http://www.geosparc.com/, Belgium.
 *
 * The program is available in open source according to the GNU Affero
 * General Public License. All contributions in this program are covered
 * by the Geomajas Contributors License Agreement. For full licensing
 * details, see LICENSE.txt in the project root.
 */

dojo.provide("geomajas.action.toolbar.ZoomOutAction");
dojo.require("geomajas.action.ToolbarAction");

dojo.declare("ZoomOutAction", ToolbarAction, {

	/**
	 * @fileoverview Zoom out (DynamicToolbar).
	 * @class This toolbar action will zoom out on click.
	 * @author Pieter De Graef
	 *
	 * @constructor
	 * @extends ToolbarAction
	 * @param id Unique identifier
	 * @param mapView The MapView object behind the MapWidget.
	 * @param delta The delta value by which to scale. Should be smaller then 1.
	 */
	constructor : function (id, mapView, delta) {
		/** Unique identifier. */
		this.id = id;

		/** This tool's image representation. */
		this.image = "zoomOutIcon";

		/** Reference to the tooltip. */
		this.tooltip = this.tooltipLocale.ZoomOutAction;
		
		/** MapView object to zoom on. */
		this.mapView = mapView;
		if (delta) {
			this.delta = parseFloat(delta);
		} else {
			this.delta = 0.5;
		}
	},

	/**
	 * The action's execution function. This function is called when the
	 * action's button is clicked. It changes the zoom level of the mapView 
	 * object.
	 * @param event Standard browser mouseevent.
	 */
	actionPerformed : function (event) {
		this.mapView.scale(this.delta, geomajas.ZoomOption.ZOOM_OPTION_LEVEL_CHANGE);
	}
});

File: client/geomajas-client-smartgwt/client/src/main/java/org/geomajas/smartgwt/client/action/toolbar/ZoomOutAction.java
/*
 * This is part of Geomajas, a GIS framework, http://www.geomajas.org/.
 *
 * Copyright 2008-2013 Geosparc nv, http://www.geosparc.com/, Belgium.
 *
 * The program is available in open source according to the GNU Affero
 * General Public License. All contributions in this program are covered
 * by the Geomajas Contributors License Agreement. For full licensing
 * details, see LICENSE.txt in the project root.
 */
package org.geomajas.smartgwt.client.action.toolbar;

import org.geomajas.smartgwt.client.action.ConfigurableAction;
import org.geomajas.smartgwt.client.action.ToolbarAction;
import org.geomajas.smartgwt.client.i18n.I18nProvider;
import org.geomajas.smartgwt.client.util.WidgetLayout;
import org.geomajas.smartgwt.client.widget.MapWidget;

import com.smartgwt.client.widgets.events.ClickEvent;
import org.geomajas.smartgwt.client.map.MapView;

/**
 * Tool which allows zooming out directly.
 * 
 * @author Pieter De Graef
 */
public class ZoomOutAction extends ToolbarAction implements ConfigurableAction {

	private MapWidget mapWidget;

	private double zoomFactor = 0.5;

	public ZoomOutAction(MapWidget mapWidget) {
		super(WidgetLayout.iconZoomOut, I18nProvider.getToolbar().zoomOutTitle(), I18nProvider
				.getToolbar().zoomOutTooltip());
		this.mapWidget = mapWidget;
	}

	public void onClick(ClickEvent event) {
		mapWidget.getMapModel().getMapView().scale(zoomFactor, MapView.ZoomOption.LEVEL_CHANGE);
	}

	public void configure(String key, String value) {
		if ("delta".equals(key)) {
			zoomFactor = Double.parseDouble(value);
		}
	}
}
